### PR TITLE
feat(exposure-driven): proxy route creation is now driven by api expo…

### PR DESCRIPTION
**Pull request by julius-malcovsky**
Refactor: Exposure-Driven Proxy Route Creation

  Changes

  Moved proxy route creation responsibility from ApiSubscription to ApiExposure. The exposure now creates all proxy routes (for subscriber zones and failover zones) and manages their lifecycle via a janitor
  pattern.

  Benefits

  - Deduplication: Multiple subscriptions in the same zone now share a single proxy route
  - Simplified logic: Single source of truth for proxy route creation
  - Better lifecycle management: Routes are cleaned up when no longer needed

  Testing

  Added 6 comprehensive test scenarios covering edge cases:
  1. Approval denial/revocation cleanup
  2. Subscriber failover to provider failover zone (no duplicates)
  3. Same-zone subscription with cross-zone failover
  4. Multiple subscriptions sharing zone with different failover configs
  5. All subscriptions in same zone (no unnecessary proxies)
  6. Provider failover zone reuse by subscribers

  Split test file for better organization:
  - apisubscription_controller_test.go - main subscription tests
  - apisubscription_controller_failover_test.go - failover scenarios

### CODE REVIEW

File: api/internal/controller/apiexposure_controller.go [hunk 1]
```diff
@@ -64,9 +64,12 @@ func (r *ApiExposureReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			handler.EnqueueRequestsFromMapFunc(r.MapApiExposureToApiExposure),
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
 		).
+		// Watch ApiSubscription with ResourceVersionChangedPredicate (not GenerationChangedPredicate)
+		// because we need to react to approval status changes, which update Status (not Spec).
+		// When a subscription is approved, ApiExposure needs to reconcile to create/update proxy routes.
 		Watches(&apiv1.ApiSubscription{},
 			handler.EnqueueRequestsFromMapFunc(r.MapApiSubscriptionToApiExposure),
-			builder.WithPredicates(predicate.Or(predicate.GenerationChangedPredicate{}, cc.DeleteOnlyPredicate{})),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
```
Comment (lukas016) [inline api/internal/controller/apiexposure_controller.go:72]: optional: this can be optimize with custom update predicate

File: api/internal/handler/util/getters.go [hunk 2]
```diff
@@ -197,3 +197,91 @@ func FindActiveAPIExposure(ctx context.Context, apiBasePath string) (bool, *apiv
...
+			logger.V(1).Info("Adding zone for proxy route", "zone", zone.Name, "reason", reason)
+		}
+	}
+
+	for i := range subList.Items {
+		sub := &subList.Items[i]
+
+		// Skip subscriptions that are being deleted; their finalizer may still
+		// be running, but they should no longer influence proxy route creation.
+		if sub.GetDeletionTimestamp() != nil {
```
Comment (lukas016) [inline api/internal/handler/util/getters.go:251]: optional: i think it is recommended use IsZero() and it should cover nil: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go#L61
